python/test/unit/language/test_core.py

```diff
@@ -4220,6 +4220,38 @@ def mul_add(data):
     assert found_fma == enable_fp_fusion
 
 
+# -----------------------
+# test propagate_nan
+# -----------------------
+
+
+@pytest.mark.parametrize("propagate_nan", ['tl.PropagateNan.NONE', 'tl.PropagateNan.ALL'])
+@pytest.mark.parametrize("func", ['tl.minimum', 'tl.maximum'])
+def test_propagate_nan(propagate_nan, func):
+    @triton.jit
+    def kernel(A, B, C):
+        tl.store(C, FUNC (tl.load(A), tl.load(B), propagate_nan=PROPAGATE_NAN))
+
+    kernel = patch_kernel(kernel, {'FUNC': func, 'PROPAGATE_NAN': propagate_nan})
+
+    for mode in ['A', 'B', 'both']:
+        A = torch.randn((1, ), device='cuda', dtype=torch.float32)
+        if mode == 'A' or mode == 'both': A[0] = torch.nan
+        B = torch.randn((1, ), device='cuda', dtype=torch.float32)
+        if mode == 'B' or mode == 'both': B[0] = torch.nan
+        C = torch.zeros_like(A, device='cuda', dtype=torch.float32)
+        kernel[(1, )](A, B, C)
+
+        if mode == 'both':
+            assert torch.isnan(C[0])
+        else:
+            if eval(propagate_nan) == tl.PropagateNan.NONE:
+                assert not torch.isnan(C[0])
+            else:
+                assert eval(propagate_nan) == tl.PropagateNan.ALL
+                assert torch.isnan(C[0])
```

Comment:
nit, maybe something like this would be simpler?:
```
if propagate_nan == 'tl.PropagateNan.ALL' or mode == 'both':
    assert torch.isnan(C[0])
else:
    assert not torch.isnan(C[0])
```